### PR TITLE
CHDEXCHAE-64 add index listener on models added

### PR DESCRIPTION
**Pull request by duttonw**
pull in the YTP parts from https://github.com/qld-gov-au/ckanext-data-qld/pull/201/files to the plugin that it's referencing in allowing index to be updated on package update for comments.

### CODE REVIEW

File: ckanext/ytp/comments/listeners.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import logging
+
+import ckan.plugins.toolkit as tk
+import ckan.lib.search as search
```
Comment (ThrawnCA) [inline ckanext/ytp/comments/listeners.py:4]: This is fine, but `from ckan.lib import search` is simpler and works better if we end up importing multiple modules from `ckan.lib`.

Comment (duttonw) [reply ckanext/ytp/comments/listeners.py:4]: unsure, was just duplicating the original, tho it seems tests don't pass :/

File: ckanext/ytp/comments/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
+# encoding: utf-8
+
+import os
+import six
+from datetime import datetime as dt
+
+import pytest
+import factory
+from faker import Faker
```
Comment (ThrawnCA) [inline ckanext/ytp/comments/tests/conftest.py:9]: This won't work on CKAN 2.8, need to either wrap this in a check or drop CKAN 2.8 support.

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

**Comment by ThrawnCA** [comment]
The new tests can't run on CKAN 2.8 because it doesn't provide the dependencies they need. The tests are deliberately disabled on CKAN 2.9. And they run, but fail, on CKAN 2.10, indicating that the index is not being updated as intended.

**Review by ThrawnCA — APPROVED** [review]

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -31,6 +31,11 @@ jobs:
         run: bin/test.sh
```
Comment (duttonw) [inline .github/workflows/test.yml:31]: please split test.sh into unit-test.sh and bdd-test.sh so its easier to find test results etc.

File: bin/test.sh [hunk 4]
```diff
@@ -11,4 +11,5 @@ echo "==> Run Unit tests"
 ahoy test-unit
 
 echo "==> Run BDD tests"
-ahoy test-bdd || (ahoy logs; exit 1)
```
Comment (duttonw) [inline bin/test.sh:14]: glad you migrated this into another step.

File: .github/workflows/test.yml (hunk 3 above)
Comment (ThrawnCA) [reply .github/workflows/test.yml]: Done, I just invoked the ahoy steps directly.

**Pull request merged**